feat(docs): Add documentation for React Hook Form usage

WillFelisberto opened the pull request:
This commit introduces a comprehensive guide on using React Hook Form, covering installation, form creation, validation with Yup, and best practices. The documentation aims to simplify the understanding and implementation of forms in React applications.

#134

Commit:
feat(docs): Add documentation for React Hook Form usage

This commit introduces a comprehensive guide on using React Hook Form, covering installation, form creation, validation with Yup, and best practices. The documentation aims to simplify the understanding and implementation of forms in React applications.

gabrielduete commented on docs/Formulários/1-React-Hook-Form.md:
```diff
@@ -0,0 +1,221 @@
+## Links úteis
+
+* [Documentação oficial](https://react-hook-form.com)
+* [HookForm + Yup Integration](https://react-hook-form.com/get-started#SchemaValidation)
```
:copilot: 

No, the link is correct

gabrielduete commented on docs/Formulários/1-React-Hook-Form.md:
```diff
@@ -0,0 +1,221 @@
+## Boas práticas
+
+* Sempre defina mensagens de erro personalizadas.
+* Use *defaultValues* para inicializar o formulário.
```
Vale colocar o defaulValues em crase tbm e deixar sem o 's'?

```suggestion
* Use `defaultValue` para inicializar o formulário.
```

gabrielduete commented on docs/Formulários/1-React-Hook-Form.md:
```diff
@@ -0,0 +1,221 @@
+
+## O que é o React Hook Form?
+
+O **React Hook Form (RHF)** é uma biblioteca leve e performática para manipulação de formulários em aplicações **React**.
```
Vale linkar a doc aqui

```suggestion
O **[React Hook Form (RHF)](https://react-hook-form.com/)** é uma biblioteca leve e performática para manipulação de formulários em aplicações **React**.
```

gabrielduete commented on docs/Formulários/1-React-Hook-Form.md:
```diff
@@ -0,0 +1,221 @@
+
+---
+
+## Links úteis
```
Muito bom!

gabrielduete commented on docs/Formulários/1-React-Hook-Form.md:
```diff
@@ -0,0 +1,221 @@
+O **React Hook Form (RHF)** é uma biblioteca leve e performática para manipulação de formulários em aplicações **React**.
+Seu principal objetivo é simplificar a criação, validação e submissão de formulários, reduzindo a necessidade de *boilerplate* e melhorando a performance através do uso dos **React Hooks**.
+
+Ao invés de controlar cada campo de formulário através de *states* e *handlers* (como em `onChange` e [`useState`](../Iniciando%20com%20React/8.1-useState.md)), o React Hook Form utiliza **referências diretas ([refs](../Iniciando%20com%20React/Refs.md))** para registrar e monitorar os campos, tornando o código mais limpo e eficiente.
```
Belo trampo em linkar outros artigos do repo! 💜

gabrielduete commented:
Muito obrigado por contribuir! 💜

gabrielduete commented:
Hey, @WillFelisberto ! Tudo bem?

Só pingando para ver se ainda tem interesse em prosseguir com a PR. Só para garantir que não esqueceu ou algo do tipo. =)

gabrielduete commented:
Hey, @WillFelisberto ! Tudo bem?

Só pingando para ver se ainda tem interesse em prosseguir com a PR. Só para garantir que não esqueceu ou algo do tipo. =)

Commit:
docs(react-hook-form): adicionado link oficial e ajuste na formatação de defaultValues

- Adicionado link para o site oficial do React Hook Form na primeira menção;
- Alterada formatação de defaultValues de itálico para código;

gabrielduete reviewed (approved):
@WillFelisberto Boa, monstro!!

Aprovado! Vlw demais por contribuir! 💜

gvieira18 commented:
> Nooo :/, agora que vi que perdi as permissões. 😢
> 
> @danielhe4rt, @rosajsx, @he4rt/4noobs-admin, @he4rt/4noobs-devs, @he4rt/escritores-tech
> 
> Podem dar uma olhada aqui? Tudo ok pelo que vi



> Parabéns pelo conteúdo, primo! Ficou sensacional! Infelizmente eu acredito que não possua mais permissão pra mergear, vamo precisar ver com o pessoal da comunidade.

O acesso direto foi removido mesmo, ambos estavam como mantenedores do repo? @gabrielduete @rosajsx

gabrielduete committed:
Merge branch 'master' into feature/react-hook-form
